man/man3/Session.3
.TH "Session" 3 "Tue Apr 14 2015" "Version 1.0" "VirtualSCADA" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Session \- 
.SH SYNOPSIS
.br
.PP
.PP
Inherits \fBSession\fP\&.
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static \fBgetSessionConfig\fP ()"
.br
.ti -1c
.RI "static \fBgetDefaultDriver\fP ()"
.br
.ti -1c
.RI "static \fBsetDefaultDriver\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBdriver\fP ($driver=null)"
.br
.ti -1c
.RI "static \fBextend\fP ($\fBdriver\fP, $\fBcallback\fP)"
.br
.ti -1c
.RI "static \fBgetDrivers\fP ()"
.br
.ti -1c
.RI "static \fBstart\fP ()"
.br
.ti -1c
.RI "static \fBgetId\fP ()"
.br
.ti -1c
.RI "static \fBsetId\fP ($id)"
.br
.ti -1c
.RI "static \fBisValidId\fP ($id)"
.br
.ti -1c
.RI "static \fBgetName\fP ()"
.br
.ti -1c
.RI "static \fBsetName\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBinvalidate\fP ($lifetime=null)"
.br
.ti -1c
.RI "static \fBmigrate\fP ($destroy=false, $lifetime=null)"
.br
.ti -1c
.RI "static \fBregenerate\fP ($destroy=false)"
.br
.ti -1c
.RI "static \fBsave\fP ()"
.br
.ti -1c
.RI "static \fBageFlashData\fP ()"
.br
.ti -1c
.RI "static \fBhas\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBget\fP ($\fBname\fP, $default=null)"
.br
.ti -1c
.RI "static \fBpull\fP ($key, $default=null)"
.br
.ti -1c
.RI "static \fBhasOldInput\fP ($key=null)"
.br
.ti -1c
.RI "static \fBgetOldInput\fP ($key=null, $default=null)"
.br
.ti -1c
.RI "static \fBset\fP ($\fBname\fP, $value)"
.br
.ti -1c
.RI "static \fBput\fP ($key, $value=null)"
.br
.ti -1c
.RI "static \fBpush\fP ($key, $value)"
.br
.ti -1c
.RI "static \fBflash\fP ($key, $value)"
.br
.ti -1c
.RI "static \fBflashInput\fP ($value)"
.br
.ti -1c
.RI "static \fBreflash\fP ()"
.br
.ti -1c
.RI "static \fBkeep\fP ($\fBkeys\fP=null)"
.br
.ti -1c
.RI "static \fBall\fP ()"
.br
.ti -1c
.RI "static \fBreplace\fP ($attributes)"
.br
.ti -1c
.RI "static \fBremove\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBforget\fP ($key)"
.br
.ti -1c
.RI "static \fBclear\fP ()"
.br
.ti -1c
.RI "static \fBflush\fP ()"
.br
.ti -1c
.RI "static \fBisStarted\fP ()"
.br
.ti -1c
.RI "static \fBregisterBag\fP ($bag)"
.br
.ti -1c
.RI "static \fBgetBag\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBgetMetadataBag\fP ()"
.br
.ti -1c
.RI "static \fBgetBagData\fP ($\fBname\fP)"
.br
.ti -1c
.RI "static \fBtoken\fP ()"
.br
.ti -1c
.RI "static \fBgetToken\fP ()"
.br
.ti -1c
.RI "static \fBregenerateToken\fP ()"
.br
.ti -1c
.RI "static \fBsetExists\fP ($value)"
.br
.ti -1c
.RI "static \fBgetHandler\fP ()"
.br
.ti -1c
.RI "static \fBhandlerNeedsRequest\fP ()"
.br
.ti -1c
.RI "static \fBsetRequestOnHandler\fP ($request)"
.br
.in -1c
.SS "Additional Inherited Members"
.SH "Detailed Description"
.PP 
Definition at line 10853 of file _ide_helper\&.php\&.
.SH "Member Function Documentation"
.PP 
.SS "static ageFlashData ()\fC [static]\fP"
Age the flash data for the session\&.
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11054 of file _ide_helper\&.php\&.
.SS "static all ()\fC [static]\fP"
Returns attributes\&.
.PP
\fBReturns:\fP
.RS 4
array Attributes  
.RE
.PP

.PP
Definition at line 11205 of file _ide_helper\&.php\&.
.SS "static clear ()\fC [static]\fP"
Clears all attributes\&.
.PP
Definition at line 11248 of file _ide_helper\&.php\&.
.SS "static driver ( $driver = \fCnull\fP)\fC [static]\fP"
Get a driver instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$driver\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 10893 of file _ide_helper\&.php\&.
.SS "static extend ( $driver,  $callback)\fC [static]\fP"
Register a custom driver creator Closure\&.
.PP
\fBParameters:\fP
.RS 4
\fI$driver\fP 
.br
\fI$callback\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
$this 
.RE
.PP

.PP
Definition at line 10906 of file _ide_helper\&.php\&.
.SS "static flash ( $key,  $value)\fC [static]\fP"
Flash a key / value pair to the session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.br
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11162 of file _ide_helper\&.php\&.
.SS "static flashInput ( $value)\fC [static]\fP"
Flash an input array to the session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11173 of file _ide_helper\&.php\&.
.SS "static flush ()\fC [static]\fP"
Remove all of the items from the session\&.
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11258 of file _ide_helper\&.php\&.
.SS "static forget ( $key)\fC [static]\fP"
Remove an item from the session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11238 of file _ide_helper\&.php\&.
.SS "static get ( $name,  $default = \fCnull\fP)\fC [static]\fP"
Returns an attribute\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP The attribute name 
.br
\fI$default\fP The default value if not found\&. 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed  
.RE
.PP

.PP
Definition at line 11079 of file _ide_helper\&.php\&.
.SS "static getBag ( $name)\fC [static]\fP"
Gets a bag instance by name\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.PP
Definition at line 11289 of file _ide_helper\&.php\&.
.SS "static getBagData ( $name)\fC [static]\fP"
Get the raw bag data array for a given bag\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
array 
.RE
.PP

.PP
Definition at line 11310 of file _ide_helper\&.php\&.
.SS "static getDefaultDriver ()\fC [static]\fP"
Get the default session driver name\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.PP
Definition at line 10871 of file _ide_helper\&.php\&.
.SS "static getDrivers ()\fC [static]\fP"
Get all of the created 'drivers'\&.
.PP
\fBReturns:\fP
.RS 4
array 
.RE
.PP

.PP
Definition at line 10917 of file _ide_helper\&.php\&.
.SS "static getHandler ()\fC [static]\fP"
Get the underlying session handler implementation\&.
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.PP
Definition at line 11361 of file _ide_helper\&.php\&.
.SS "static getId ()\fC [static]\fP"
Returns the session ID\&.
.PP
\fBReturns:\fP
.RS 4
string The session ID\&.  
.RE
.PP

.PP
Definition at line 10941 of file _ide_helper\&.php\&.
.SS "static getMetadataBag ()\fC [static]\fP"
Gets session meta\&.
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.PP
Definition at line 11299 of file _ide_helper\&.php\&.
.SS "static getName ()\fC [static]\fP"
Returns the session name\&.
.PP
\fBReturns:\fP
.RS 4
mixed The session name\&.  
.RE
.PP

.PP
Definition at line 10974 of file _ide_helper\&.php\&.
.SS "static getOldInput ( $key = \fCnull\fP,  $default = \fCnull\fP)\fC [static]\fP"
Get the requested item from the flashed input array\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.br
\fI$default\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 11114 of file _ide_helper\&.php\&.
.SS "static getSessionConfig ()\fC [static]\fP"
Get the session configuration\&.
.PP
\fBReturns:\fP
.RS 4
array 
.RE
.PP

.PP
Definition at line 10861 of file _ide_helper\&.php\&.
.SS "static getToken ()\fC [static]\fP"
Get the CSRF token value\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.PP
Definition at line 11330 of file _ide_helper\&.php\&.
.SS "static handlerNeedsRequest ()\fC [static]\fP"
Determine if the session handler needs a request\&.
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 11371 of file _ide_helper\&.php\&.
.SS "static has ( $name)\fC [static]\fP"
Checks if an attribute is defined\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP The attribute name 
.RE
.PP
\fBReturns:\fP
.RS 4
bool true if the attribute is defined, false otherwise  
.RE
.PP

.PP
Definition at line 11066 of file _ide_helper\&.php\&.
.SS "static hasOldInput ( $key = \fCnull\fP)\fC [static]\fP"
Determine if the session contains old input\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 11102 of file _ide_helper\&.php\&.
.SS "static invalidate ( $lifetime = \fCnull\fP)\fC [static]\fP"
Invalidates the current session\&.
.PP
Clears all session attributes and flashes and regenerates the session and deletes the old session from persistence\&.
.PP
\fBParameters:\fP
.RS 4
\fI$lifetime\fP Sets the cookie lifetime for the session cookie\&. \fBA\fP null value will leave the system settings unchanged, 0 sets the cookie to expire with browser session\&. Time is in seconds, and is not a Unix timestamp\&. 
.RE
.PP
\fBReturns:\fP
.RS 4
bool True if session invalidated, false if error\&.  
.RE
.PP

.PP
Definition at line 11003 of file _ide_helper\&.php\&.
.SS "static isStarted ()\fC [static]\fP"
Checks if the session was started\&.
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 11268 of file _ide_helper\&.php\&.
.SS "static isValidId ( $id)\fC [static]\fP"
Determine if this is a valid session ID\&.
.PP
\fBParameters:\fP
.RS 4
\fI$id\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 10963 of file _ide_helper\&.php\&.
.SS "static keep ( $keys = \fCnull\fP)\fC [static]\fP"
Reflash a subset of the current flash data\&.
.PP
\fBParameters:\fP
.RS 4
\fI$keys\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11194 of file _ide_helper\&.php\&.
.SS "static migrate ( $destroy = \fCfalse\fP,  $lifetime = \fCnull\fP)\fC [static]\fP"
Migrates the current session to a new session id while maintaining all session attributes\&.
.PP
\fBParameters:\fP
.RS 4
\fI$destroy\fP Whether to delete the old session or leave it to garbage collection\&. 
.br
\fI$lifetime\fP Sets the cookie lifetime for the session cookie\&. \fBA\fP null value will leave the system settings unchanged, 0 sets the cookie to expire with browser session\&. Time is in seconds, and is not a Unix timestamp\&. 
.RE
.PP
\fBReturns:\fP
.RS 4
bool True if session migrated, false if error\&.  
.RE
.PP

.PP
Definition at line 11020 of file _ide_helper\&.php\&.
.SS "static pull ( $key,  $default = \fCnull\fP)\fC [static]\fP"
Get the value of a given key and then forget it\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.br
\fI$default\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 11091 of file _ide_helper\&.php\&.
.SS "static push ( $key,  $value)\fC [static]\fP"
Push a value onto a session array\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.br
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11150 of file _ide_helper\&.php\&.
.SS "static put ( $key,  $value = \fCnull\fP)\fC [static]\fP"
Put a key / value pair or array of key / value pairs in the session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$key\fP 
.br
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11138 of file _ide_helper\&.php\&.
.SS "static reflash ()\fC [static]\fP"
Reflash all of the session flash data\&.
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11183 of file _ide_helper\&.php\&.
.SS "static regenerate ( $destroy = \fCfalse\fP)\fC [static]\fP"
Generate a new session identifier\&.
.PP
\fBParameters:\fP
.RS 4
\fI$destroy\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 11031 of file _ide_helper\&.php\&.
.SS "static regenerateToken ()\fC [static]\fP"
Regenerate the CSRF token value\&.
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11340 of file _ide_helper\&.php\&.
.SS "static registerBag ( $bag)\fC [static]\fP"
Registers a SessionBagInterface with the session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$bag\fP 
.RE
.PP

.PP
Definition at line 11278 of file _ide_helper\&.php\&.
.SS "static remove ( $name)\fC [static]\fP"
Removes an attribute\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed The removed value or null when it does not exist  
.RE
.PP

.PP
Definition at line 11227 of file _ide_helper\&.php\&.
.SS "static replace ( $attributes)\fC [static]\fP"
Sets attributes\&.
.PP
\fBParameters:\fP
.RS 4
\fI$attributes\fP Attributes 
.RE
.PP

.PP
Definition at line 11215 of file _ide_helper\&.php\&.
.SS "static save ()\fC [static]\fP"
Force the session to be saved and closed\&.
.PP
This method is generally not required for real sessions as the session will be automatically saved at the end of code execution\&. 
.PP
Definition at line 11044 of file _ide_helper\&.php\&.
.SS "static set ( $name,  $value)\fC [static]\fP"
Sets an attribute\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.br
\fI$value\fP 
.RE
.PP

.PP
Definition at line 11126 of file _ide_helper\&.php\&.
.SS "static setDefaultDriver ( $name)\fC [static]\fP"
Set the default session driver name\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 10882 of file _ide_helper\&.php\&.
.SS "static setExists ( $value)\fC [static]\fP"
Set the existence of the session on the handler if applicable\&.
.PP
\fBParameters:\fP
.RS 4
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11351 of file _ide_helper\&.php\&.
.SS "static setId ( $id)\fC [static]\fP"
Sets the session ID
.PP
\fBParameters:\fP
.RS 4
\fI$id\fP 
.RE
.PP

.PP
Definition at line 10952 of file _ide_helper\&.php\&.
.SS "static setName ( $name)\fC [static]\fP"
Sets the session name\&.
.PP
\fBParameters:\fP
.RS 4
\fI$name\fP 
.RE
.PP

.PP
Definition at line 10985 of file _ide_helper\&.php\&.
.SS "static setRequestOnHandler ( $request)\fC [static]\fP"
Set the request on the handler instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$request\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 11382 of file _ide_helper\&.php\&.
.SS "static start ()\fC [static]\fP"
Starts the session storage\&.
.PP
\fBReturns:\fP
.RS 4
bool True if session started\&. 
.RE
.PP
\fBExceptions:\fP
.RS 4
\fI\fP .RE
.PP

.PP
Definition at line 10930 of file _ide_helper\&.php\&.
.SS "static token ()\fC [static]\fP"
Get the CSRF token value\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.PP
Definition at line 11320 of file _ide_helper\&.php\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for VirtualSCADA from the source code\&.


virtualscada_laravel/config/session.php
<?php

return [

	/*
	|--------------------------------------------------------------------------
	| Default Session Driver
	|--------------------------------------------------------------------------
	|
	| This option controls the default session "driver" that will be used on
	| requests. By default, we will use the lightweight native driver but
	| you may specify any of the other wonderful drivers provided here.
	|
	| Supported: "file", "cookie", "database", "apc",
	|            "memcached", "redis", "array"
	|
	*/
t
	'driver' => env('SESSION_DRIVER', 'file'),

	/*
	|--------------------------------------------------------------------------
	| Session Lifetime
	|--------------------------------------------------------------------------
	|
	| Here you may specify the number of minutes that you wish the session
	| to be allowed to remain idle before it expires. If you want them
	| to immediately expire on the browser closing, set that option.
	|
	*/

	'lifetime' => 1,

	'expire_on_close' => true,

	/*
	|--------------------------------------------------------------------------
	| Session Encryption
	|--------------------------------------------------------------------------
	|
	| This option allows you to easily specify that all of your session data
	| should be encrypted before it is stored. All encryption will be run
	| automatically by Laravel and you can use the Session like normal.
	|
	*/

	'encrypt' => false,

	/*
	|--------------------------------------------------------------------------
	| Session File Location
	|--------------------------------------------------------------------------
	|
	| When using the native session driver, we need a location where session
	| files may be stored. A default has been set for you but a different
	| location may be specified. This is only needed for file sessions.
	|
	*/

	'files' => storage_path().'/framework/sessions',

	/*
	|--------------------------------------------------------------------------
	| Session Database Connection
	|--------------------------------------------------------------------------
	|
	| When using the "database" or "redis" session drivers, you may specify a
	| connection that should be used to manage these sessions. This should
	| correspond to a connection in your database configuration options.
	|
	*/

	'connection' => null,

	/*
	|--------------------------------------------------------------------------
	| Session Database Table
	|--------------------------------------------------------------------------
	|
	| When using the "database" session driver, you may specify the table we
	| should use to manage the sessions. Of course, a sensible default is
	| provided for you; however, you are free to change this as needed.
	|
	*/

	'table' => 'sessions',

	/*
	|--------------------------------------------------------------------------
	| Session Sweeping Lottery
	|--------------------------------------------------------------------------
	|
	| Some session drivers must manually sweep their storage location to get
	| rid of old sessions from storage. Here are the chances that it will
	| happen on a given request. By default, the odds are 2 out of 100.
	|
	*/

	'lottery' => [2, 100],

	/*
	|--------------------------------------------------------------------------
	| Session Cookie Name
	|--------------------------------------------------------------------------
	|
	| Here you may change the name of the cookie used to identify a session
	| instance by ID. The name specified here will get used every time a
	| new session cookie is created by the framework for every driver.
	|
	*/

	'cookie' => 'laravel_session',

	/*
	|--------------------------------------------------------------------------
	| Session Cookie Path
	|--------------------------------------------------------------------------
	|
	| The session cookie path determines the path for which the cookie will
	| be regarded as available. Typically, this will be the root path of
	| your application but you are free to change this when necessary.
	|
	*/

	'path' => '/',

	/*
	|--------------------------------------------------------------------------
	| Session Cookie Domain
	|--------------------------------------------------------------------------
	|
	| Here you may change the domain of the cookie used to identify a session
	| in your application. This will determine which domains the cookie is
	| available to in your application. A sensible default has been set.
	|
	*/

	'domain' => null,

	/*
	|--------------------------------------------------------------------------
	| HTTPS Only Cookies
	|--------------------------------------------------------------------------
	|
	| By setting this option to true, session cookies will only be sent back
	| to the server if the browser has a HTTPS connection. This will keep
	| the cookie from being sent to you if it can not be done securely.
	|
	*/

	'secure' => false,

];
